qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -4,6 +4,59 @@
 from qcodes import VisaInstrument
 from qcodes.instrument.parameter import ArrayParameter
 from qcodes.utils.validators import Numbers, Ints, Enum, Strings
+from qcodes.instrument.channel import InstrumentChannel, ChannelList
+
+
+class SR830AuxArray(InstrumentChannel):  # We see that InstrumentChannel, ChannelList are something of a misnomer
+    def __init__(self, parent, name, aux_number):
+        super().__init__(parent, name)
+
+        self.add_parameter('aux_in',
+                           label='Aux input {}'.format(aux_number),
+                           get_cmd='OAUX? {}'.format(aux_number),
+                           get_parser=float,
+                           unit='V')
+
+        self.add_parameter('aux_out',
+                           label='Aux output {}'.format(aux_number),
+                           get_cmd='AUXV? {}'.format(aux_number),
+                           get_parser=float,
+                           set_cmd='AUXV {0}, {{}}'.format(aux_number),
+                           unit='V')
+
+
+class SR830ChannelArray(InstrumentChannel):
+    def __init__(self, parent, name, channel_number):
+        super().__init__(parent, name)
+
+        self._parent = parent
+
+        # detailed validation and mapping performed in set/get functions. Note that we only have
+        # two channels
+        self.add_parameter(
+            'ratio',
+            label='Channel {} ratio'.format(channel_number),
+            get_cmd=partial(parent._get_ch_ratio, channel_number),
+            set_cmd=partial(parent._set_ch_ratio, channel_number),
+            vals=Strings()
+        )
+
+        self.add_parameter(
+            'display',
+            label='Channel {} display'.format(channel_number),
+            get_cmd=partial(parent._get_ch_display, channel_number),
+            set_cmd=partial(parent._set_ch_display, channel_number),
+            vals=Strings()
+        )
+
+        self.add_parameter(
+            'databuffer',
+            channel=channel_number,
+            parameter_class=ChannelBuffer
+        )
+
+        def get_parent():
```

Comment:
A public accessor is probably a good idea. But I would make it a property and add it to the Super class